pkg/kube/ready.go

```diff
@@ -300,7 +300,7 @@ func (c *ReadyChecker) daemonSetReady(ds *appsv1.DaemonSet) bool {
 	}
 
 	expectedReady := int(ds.Status.DesiredNumberScheduled) - maxUnavailable
-	if !(int(ds.Status.NumberReady) >= expectedReady) {
+	if !(int(ds.Status.NumberReady) >= expectedReady) || expectedReady < 0 {
```

Comment:
`DesiredNumberScheduled` is not calculated until it's taken off the queue and processed. There is a potential for a race condition here. If the `maxUnavailable > 0` and `DesiredNumberScheduled = 0` then `NumberReady` of 0 would be > `expectedReady` (which would be negative as 0 - maxUnavailable). This case would fall in to the `if` statement.

But, if `maxUnavailable` is set to 0 you have 0 desired number scheduled - 0 max unavailable equaling 0 or more as the number ready.

I don't think this completely fixes the bug because it shows up when `expectedReady` is 0. As in both `DesiredNumberScheduled` and `maxUnavailable` are 0.